### PR TITLE
Migrate all static AddAction extensions to dynamic extensions

### PR DESCRIPTION
**Pull request by rohitkrai03**
Depends on https://github.com/openshift/console/pull/8363

**Fixes**: https://issues.redhat.com/browse/ODC-5550

**Analysis / Root cause**: All `AddAction` extensions currently use static extensibility mechanism.

**Solution Description**: Move all static AddAction extensions to use dynamic extensibility.

**Screen shots / Gifs for design review**: No UI change.

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts
Comment (vojtechszocs) [inline frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts:36]: Do we need `R` type param here?

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts
Comment (vojtechszocs) [inline frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts:26]: Do we need `T` type param here?

**Comment by christianvogt** [comment]
@rohitkrai03 doesn't https://github.com/openshift/console/pull/8363 need to merge first otherwise localization of `.json` files won't happen?

File: frontend/packages/dev-console/console-extensions.json
Comment (vojtechszocs) [inline frontend/packages/dev-console/console-extensions.json:7]: I'd suggest to wait until #8363 is merged and then remove these `// t('..')` comments in JSON files.

File: frontend/packages/dev-console/package.json
Comment (vojtechszocs) [inline frontend/packages/dev-console/package.json:26]:
```suggestion
      "icons": "src/utils/icons.tsx"
```

Can you also update `packages/kubevirt-plugin/package.json` accordingly?

File: frontend/packages/dev-console/src/components/EmptyState.tsx
Comment (vojtechszocs) [inline frontend/packages/dev-console/src/components/EmptyState.tsx:27]: Can't we just use `ResolvedAddAction` here?

**Comment by rohitkrai03** [comment]
> @rohitkrai03 doesn't #8363 need to merge first otherwise localization of `.json` files won't happen?

@christianvogt Yeah, i am trying your PR with my branch. Will rebase my branch to include your changes once everything works fine.

File: frontend/packages/dev-console/src/utils/icons.tsx
Comment (vojtechszocs) [inline frontend/packages/dev-console/src/utils/icons.tsx:9]: Check out https://github.com/tc39/proposal-export-ns-from#current-ecmascript-2015-modules

Can we do `export { gitIcon as gitIconSVG } from '../images/from-git.svg';` here, in order to cut down the complexity?

File: frontend/packages/helm-plugin/package.json
Comment (vojtechszocs) [inline frontend/packages/helm-plugin/package.json:14]:
```suggestion
      "icons": "src/utils/icons.ts"
```

File: frontend/packages/knative-plugin/package.json
Comment (vojtechszocs) [inline frontend/packages/knative-plugin/package.json:24]:
```suggestion
      "icons": "src/utils/icons.ts"
```

File: frontend/packages/kubevirt-plugin/package.json
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/package.json:46]:
```suggestion
      "icons": "src/utils/icons.tsx"
```

File: frontend/packages/kubevirt-plugin/package.json
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/package.json:45]:
```suggestion
      "standaloneConsole": "src/components/vms/vm-console/StandaloneVMConsolePage.tsx",
```

File: frontend/packages/pipelines-plugin/package.json
Comment (vojtechszocs) [inline frontend/packages/pipelines-plugin/package.json:23]:
```suggestion
      "icons": "src/utils/icons.ts"
```

**Review by vojtechszocs — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/EmptyState.tsx
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/EmptyState.tsx:27]: Since `useResolvedExtensions` returns `ResolvedExtension<E>[]` we would need to use `ResolvedExtension` type here.

File: frontend/packages/dev-console/src/utils/icons.tsx
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/utils/icons.tsx:9]: SVG modules do not use named exports so we would need something like `export { * as gitIconSVG } from '../images/from-git.svg';` which is not supported as of yet. Its a proposal for ES2021 I think https://github.com/tc39/proposal-export-ns-from#proposed-addition.

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts
Comment (rohitkrai03) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts]: Removed.

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts
Comment (rohitkrai03) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts]: Removed.

File: frontend/packages/dev-console/console-extensions.json
Comment (rohitkrai03) [reply frontend/packages/dev-console/console-extensions.json]: Removed the comment and rebased with https://github.com/openshift/console/pull/8363.

File: frontend/packages/dev-console/package.json
Comment (rohitkrai03) [reply frontend/packages/dev-console/package.json]: Done.

File: frontend/packages/helm-plugin/package.json
Comment (rohitkrai03) [reply frontend/packages/helm-plugin/package.json]: Done.

File: frontend/packages/kubevirt-plugin/package.json
Comment (rohitkrai03) [reply frontend/packages/kubevirt-plugin/package.json]: Done.

File: frontend/packages/kubevirt-plugin/package.json
Comment (rohitkrai03) [reply frontend/packages/kubevirt-plugin/package.json]: Done.

File: frontend/packages/pipelines-plugin/package.json
Comment (rohitkrai03) [reply frontend/packages/pipelines-plugin/package.json]: Done.

File: frontend/packages/knative-plugin/package.json
Comment (rohitkrai03) [reply frontend/packages/knative-plugin/package.json]: Done.

File: frontend/packages/dev-console/src/utils/icons.tsx
Comment (vojtechszocs) [reply frontend/packages/dev-console/src/utils/icons.tsx:9]: > SVG modules do not use named exports
This is the part I was missing, thanks for clarifying.

It seems that `export * from "mod";` proposal is stage 4 (already merged to ECMA-262) and TypeScript 3.8 [should support it](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#export--as-ns-syntax) so maybe we can use it already?

Comment (vojtechszocs) [reply frontend/packages/dev-console/src/utils/icons.tsx:9]: Hmm, maybe webpack 4.x version we use [might not support it yet](https://github.com/webpack/webpack/issues/10460), so if it doesn't work, I'm perfectly fine with existing code.

Comment (vojtechszocs) [reply frontend/packages/dev-console/src/utils/icons.tsx:9]: OK never mind, this is only supported in webpack 5.x ([link](https://github.com/webpack/webpack/releases/tag/v5.0.0-beta.21)) so can't use it yet.

**Comment by vojtechszocs** [comment]
LGTM on last two commits.

File: frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts
Comment (rohitkrai03) [reply frontend/packages/console-dynamic-plugin-sdk/src/extensions/add-actions.ts]: Done.

**Pull request merged**
